README.md

```diff
@@ -16,25 +16,70 @@ reading out data from Modbus devices. The design includes:
-6. write data to output FIFO
-255. stops application loop, exit FPGA application
-
-Future (currently unsupported) instructions:
-
-* write something (transformed part of output array) to telemetry FIFO
+## Supported instructions
+
+| Instruction | Description                                             |
+| ----------- | ------------------------------------------------------- |
+| 0           | Stop executing and wait for a new instructions in FIFO. |
+| 1           | Write the next byte to the port.                        |
+| 2           | Wait for given ms.                                      |
+| 3           | Read from the port and store data into output array.    |
+| 4           | Loop (jump to some instruction, usually 0 to repeat the sequence; with  looping, a check for commanding FIFO will be provided, and if that contains new instructions, instruction stack will be replaced with those new instructions). |
+| 5           | Check readout [CRC](https://en.wikipedia.org/wiki/Cyclic_redundancy_check), assumes CRC are two last bytes read. If CRC doesn't match, error 6000 is reported and FPGA processing stops. |
+| 6           | Write data to output FIFO.                              |
+| 20          | Write multiple bytes. Followed by number of bytes and paylod (bytes to transwer). |
+| 30          | Write single byte to telemetry. Instruction shall be followed with telemetry address offset and output array offset. |
+| 31          | Write two bytes to telemetry. Instruction shall be followed with telemetry address offset and output array offset (of the first byte). |
+| 32          | Write four bytes to telemetry. Instruction shall be followed with telemetry address offset and output array offset (of the first byte). |
+| 33          | Write eight bytes to telemetry. Instruction shall be followed with telemetry address offset and output array offset (of the first byte). |
+| 255         | Stops application loop, exit FPGA application. |
+
+### Telemetry
+
+Telemetry commands (30-34) writes big endian numbers.
+
+### Example
+
+When following is passed to input FIFO, the processing unit will:
+
+1. call for function 58 from address 1, CRC is 0x3380
+2. waits 100 milliseconds for response
+3. reads 18 bytes from device
+4. check CRC
+5. writes two first bytes from payload (indices 2 and 3; first two bytes are
+   address and function) to telemetry at addrress 11
+6. writes eight bytes from payload indices 4-11 to telemetry at address 22
+7. writes four bytes (indices 12-15) to telemetry at address 67
+8. waits 500 milliseconds
+9. closes loop (call again function 58,..)
+
+The loops get repeated as long as new instructions aren't available on
+commanding FIFO, or an error occurs.
+
+0. 23
+1. 20 4 1 58 80 33
+2. 2 100
+3. 3 16
+4. 5
+5. 31 11 2
+6. 33 22 4
+7. 32 67 12
+8. 2 500
+9. 4
+
+Assuming device response is (CRC is 0x6676):
+
+1 58 0x11 0x12 0x13 0x14 0x15 0x16 0x17 0x18 0x19 0x20 0x21 0x22 0x23 0x24 0x76 0x66
```

Comment:
Please consider to add more explanation for this part.

Reply:
Those are raw data from device. I am not sure how much I can add.